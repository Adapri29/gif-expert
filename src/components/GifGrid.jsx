import { Giftem } from './Giftem';
import {useFetchGifs} from '../hooks/useFetchGifs'

export const GifGrid = ({category}) => {

    const {images, isLoading} = useFetchGifs( category )


    return (
        <> 
    
            <h3>{category}</h3>
            {
                isLoading && (<h2>Cargando...</h2>)
            }

            <div className='card-grid'>
                {
                    images.map((img) => (
                        <Giftem 
                            key={img.id} 
                            //Esparcir propiedades
                            {...img}  
                        />
                    ))
                }

            </div>
    
        </>
    )
}
