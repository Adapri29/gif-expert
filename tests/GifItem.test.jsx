import { render, screen } from "@testing-library/react"
import { GifItem } from "../src/components/GifItem"



describe('GifItem Test', () => {

    const title="Dragon Ball"
    const url= "https://songoku.jpg"


    test('Debe de hacer match con el snapshot', () => {
        
        const {container} = render(<GifItem title={title} url={url}/>)

        expect(container).toMatchSnapshot()

    })

    test('Debe mostrar la imagen con el url y alt indicado', () => { 

        render(<GifItem title={title} url={url}/>)

        // expect(screen.getByRole("img").src).toBe(url)
        // expect(screen.getByRole("img").alt).toBe(title)
        
        const {src, alt} = screen.getByRole("img")
        expect(src).toContain(url)
        expect(alt).toBe(title)

     })

     test('debe de mostrar el titulo en el componente', () => {
        render(<GifItem title={title} url={url}/>)

        expect(screen.getByText(title)).toBeTruthy()
     })
 })