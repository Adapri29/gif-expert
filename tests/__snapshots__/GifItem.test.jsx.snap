// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem Test Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Dragon Ball"
      src="https://songoku.jpg"
    />
    <p>
      Dragon Ball
    </p>
  </div>
</div>
`;
